kornia/losses/focal.py

```diff
@@ -194,27 +232,51 @@ def binary_focal_loss_with_logits(
         )
 
     KORNIA_CHECK_SHAPE(input, ["B", "C", "*"])
+    KORNIA_CHECK(input.shape == target.shape, f'Expected target size {input.shape}, got {target.shape}')
     KORNIA_CHECK(
-        input.shape[0] == target.shape[0],
-        f'Expected input batch_size ({input.shape[0]}) to match target batch_size ({target.shape[0]}).',
+        input.device == target.device,
+        f"input and target must be in the same device. Got: {input.device} and {target.device}",
     )
 
-    if pos_weight is None:
-        pos_weight = torch.ones([input.shape[1], *([1] * len(input.shape[2:]))], device=input.device, dtype=input.dtype)
-
-    KORNIA_CHECK_IS_TENSOR(pos_weight)
-    KORNIA_CHECK(input.shape[1] == pos_weight.shape[0], "Expected pos_weight equals number of classes.")
+    log_probs_pos: Tensor = nn.functional.logsigmoid(input)
+    log_probs_neg: Tensor = nn.functional.logsigmoid(-input)
+
+    pos_term: Tensor = -log_probs_neg.exp().pow(gamma) * target * log_probs_pos
```

Comment:
Splitting original into two terms should have no performance issue here. The benchmark shows no performance decrease.